src/View/View.php

```diff
@@ -707,12 +711,15 @@ public function render($view = null, $layout = null): string
             $this->dispatchEvent('View.afterRender', [$viewFileName]);
         }
 
-        if ($this->layout && $this->autoLayout) {
+        if ($this->autoLayout) {
             $this->Blocks->set('content', $this->renderLayout('', $this->layout));
         }
         if ($layout !== null) {
             $this->layout = $defaultLayout;
         }
+        if ($defaultAutoLayout !== null) {
+            $this->autoLayout = $defaultAutoLayout;
```

Comment:
If the view rendering raises an exception do we need to worry about resetting this state? Could we use local variables instead of modifying the instance state?

Reply:
One could check and/or modify this property from view file too. Of we just use a local variable one would no longer be able to turn off layout rendering from view which is possible right now by setting layout to false instead.

Also in case of an exception the exception renderer will use a new View instance isn't it? So we don't have to worry about resetting state.